### DISCUSSION TITLE
The "format" parameter on the Transceiver

### DISCUSSION THREAD

**Discussion by LuisGustavoMac** [Telecominfraproject/oopt-gnpy#434] [Q&A] [answered]
Hello everyone. 

I have a question about the "Transceiver" in the field "format".  What does "format" mean besides ID mode? 

Because, in parameters of "Transceiver", "format" is set to "mode 1", "mode 2", "mode 3" or "mode 4". What does each "mode" represents? Is that can be the type of modulation format?

I searched in documentation and I didn't find explanation about what this field means.

A example in code GNpy:

"Transceiver":[
            {
            "type_variety": "vendorA_trx-type1",
            "frequency":{
                        "min": 193e12,
                        "max": 193.5e12
                        },
            "mode":[
                       {

                       "format": "mode 1",
                       "baud_rate": 32e9,
                       "OSNR": 11,
                       "bit_rate": 100e9,
                       "roll_off": 0.15,
                       "tx_osnr": 40,
                       "min_spacing": 37.5e9,
                       "cost":1

**Comment by jktjkt** [comment]
> Is that can be the type of modulation format?

Yes. It is just a free-form identification to distinguish between different modes that the transponder supports. The actual value is not important for GNPy, its only role is an identifier for requesting or returning a given mode. Using descriptive mode names such as "200G 16QAM SD-FEC" is recommended, see, e.g., our [examples for OpenROADM](https://github.com/Telecominfraproject/oopt-gnpy/blob/v2.5/gnpy/example-data/eqpt_config_openroadm_ver5.json#L135).

**Reply by LuisGustavoMac** [reply]
Thank you for your answer @jktjkt.

I want to use differents modulation formats and I didn't find in the GNpy code, which identifier "ID mode" refers to the modulation format. So, the descriptive mode names is the general mode that I can use. Now, it's clearer this recommendation. 
Are there some other documentation that shows more the modes that the transponders supports, e.g, PM 64 QAM, 16QAM PCS?

**Reply by jktjkt** [reply, marked as answer]
Just to clarify -- GNPy itself does not care whether a signal uses 4, 16, 64 or, e.g., 256 modulation states. What matters for the actual simulation engine is the symbol rate and the roll-of shape of the spectrum. The more symbols the higher are the GSNR requirements, but that's implemented explicitly via the "minimal required GSNR" parameter for each optical mode.
